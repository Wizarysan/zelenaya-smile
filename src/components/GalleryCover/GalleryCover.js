import React from 'react';
import {Link} from 'react-router';
import ScrollReveal from 'scrollreveal';
import './GalleryCover.css';

import ArrowRight from './../../assets/svg/Arrow_right.js'
import AlbumGhost from './../../assets/svg/Album_ghost.js'
import AlbumSword from './../../assets/svg/Album_sword.js'
import AlbumLips from './../../assets/svg/Album_lips.js'
import AlbumCherry from './../../assets/svg/Album_cherry.js'
import AlbumGuitar from './../../assets/svg/Album_guitar.js'
import AlbumKitty from './../../assets/svg/Album_kitty.js'
import AlbumHeart from './../../assets/svg/Album_heart.js'

class GalleryCover extends React.Component {
  constructor(props) {
    super(props);
    this.sr = ScrollReveal();
  }

  componentDidMount() {
    let horOffset;
    (window.innerHeight < 900) ? horOffset = -300 :  horOffset = -100;
    if(this.props.revealForce) {
      this.sr.reveal('.gallery_cover__textwrap', {viewOffset: { bottom: horOffset }, afterReveal: function (domEl) {
        domEl.querySelector('.gallery_cover__icon').classList.add('active');
      }});
    }
  }

  handleSound(icon){
    document.getElementById(`sound-${icon}`).play();
  }

  chooseCoverIcon(icon) {
    switch(icon) {
      case 'ghost':
        return <AlbumGhost/>;
      case 'sword':
        return <AlbumSword/>;
      case 'lips':
        return <AlbumLips/>;
      case 'cherry':
        return <AlbumCherry/>;
      case 'guitar':
        return <AlbumGuitar/>;
      case 'kitty':
        return <AlbumKitty/>;
      case 'heart':
        return <AlbumHeart/>;
      default:
        return '';
    }
  }

  render() {
    return (
      <div className="gallery_cover text-center">
        <Link to={this.props.link} >
          <div className="text-left gallery_cover__textwrap">
            <div className="js-play-sound" onMouseEnter={()=>this.handleSound(this.props.icon)}>
              {this.chooseCoverIcon(this.props.icon)}
            </div>
            <div className="gallery_cover__hover photoshadow-grey">
              <img src={this.props.image} className="gallery_cover__image" alt=""/>
              <div className="gallery_cover__hover-green">
                <div className="gallery_cover__hover-photos center-both">
                  <div className="gallery_cover__hover-digit">{this.props.qty}</div>
                  <div className="gallery_cover__hover-text">фото</div>
                </div>
              </div>
            </div>
            <br/>
            <span className="gallery_cover__name">{this.props.name}<ArrowRight /></span>
            <br/>
            <span className="gallery_cover__date">{this.props.date}</span>
          </div>
        </Link>
      </div>
    );
  }
}

export default GalleryCover;
