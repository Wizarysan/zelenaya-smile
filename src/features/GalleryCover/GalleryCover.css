@media screen and (max-width: 767px) {
  .mobile-hide {
    display: none; }
  .mobile-show {
    display: initial; } }

@media screen and (min-width: 768px) {
  .mobile-hide {
    display: initial; }
  .mobile-show {
    display: none; } }

.animation__photo-enter {
  filter: blur(2px) grayscale(100%); }

.animation__photo-enter-active.animation__photo-enter {
  filter: blur(0px) grayscale(0%);
  transition: all 200ms ease-in; }

.animation__photo-leave {
  opacity: 1; }
  .animation__photo-leave-active.animation__photo-leave {
    opacity: 0.01;
    transition: opacity 200ms ease-in; }

@keyframes bump {
  0% {
    transform: scale(1); }
  50% {
    transform: scale(1.6); }
  100% {
    transform: scale(1); } }

.animation__bump, .gallery_cover__icon:hover {
  animation: bump .5s ease-out; }

.gallery_cover {
  position: relative; }
  .gallery_cover__image {
    max-width: 100%;
    max-height: 560px; }
  .gallery_cover__hover {
    position: relative; }
    .gallery_cover__hover-green {
      position: absolute;
      top: 0;
      width: 100%;
      height: 100%;
      background-color: rgba(115, 189, 76, 0.8);
      line-height: 0.9;
      opacity: 0;
      transition: opacity .3s; }
      .gallery_cover__hover-green:hover {
        opacity: 1; }
    .gallery_cover__hover-photos {
      color: white;
      text-align: center; }
    .gallery_cover__hover-digit {
      font-size: 80px; }
    .gallery_cover__hover-text {
      font-size: 20px; }
  .gallery_cover__textwrap {
    position: relative;
    display: inline-block;
    max-width: 100%; }
    .gallery_cover__textwrap:hover {
      color: #73bd4c;
      fill: #73bd4c; }
  .gallery_cover__icon {
    position: absolute;
    z-index: 1;
    width: 90px;
    transition: all .5s;
    transform: scale(0.4) skew(-30deg, 30deg);
    opacity: 0; }
    @media (max-width: 767px) {
      .gallery_cover__icon {
        display: none; } }
    .gallery_cover__icon.active {
      transform: scale(1) skew(0deg, 0deg);
      opacity: 1; }
  .gallery_cover__name {
    vertical-align: middle;
    font-size: 1.1rem;
    transition: color .3s; }
  .gallery_cover__date {
    transition: color .3s; }
  .gallery_cover .arrow_right {
    display: inline-block;
    transition: all .3s; }
  .gallery_cover:hover .arrow_right {
    transform: translateX(5px); }
