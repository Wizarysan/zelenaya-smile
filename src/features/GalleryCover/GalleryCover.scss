@import './../../styles/shared.scss';

.gallery_cover {
  position: relative;
  &__image {
    max-width: 100%;
    max-height: 560px;
  }
  &__hover {
    position: relative;
    &-green {
      position: absolute;
      top: 0;
      width: 100%;
      height: 100%;
      background-color: rgba(115,189,76,0.8);
      line-height: 0.9;
      opacity: 0;
      transition: opacity .3s;
      &:hover {
        opacity: 1;
      }
    }
    &-photos {
      color:white;
      text-align: center;
    }
    &-digit {
      font-size: 80px;
    }
    &-text {
      font-size:20px;
    }
  }
  &__textwrap {
    position: relative;
    display: inline-block;
    max-width: 100%;
    &:hover {
      color: $br-green;
      fill: $br-green;
    }
  }
  &__icon {
    position: absolute;
    z-index: 1;
    width: 90px;
    transition: all .5s;
    transform: scale(0.4) skew(-30deg, 30deg);
    opacity: 0;
    @include breakpoint(xs) {
      display: none;
    }
    &.active {
      transform: scale(1) skew(0deg, 0deg);
      opacity: 1;
    }
    &:hover {
      @extend .animation__bump;
    }
  }
  &__name {
    vertical-align: middle;
    font-size: 1.1rem;
    transition: color .3s;
  }
  &__date {
    transition: color .3s;
  }
  .arrow_right {
    display: inline-block;
    transition: all .3s;
  }
  &:hover {
    .arrow_right {
      transform: translateX(5px);
    }
  }
}
