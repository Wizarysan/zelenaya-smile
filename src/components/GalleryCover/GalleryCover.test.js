import React from 'react';
import GalleryCover from './GalleryCover.js';
import renderer from 'react-test-renderer';

it('renders correctly with test props', () => {
  const tree = renderer
    .create(<GalleryCover link="testLink" currentId="1" currentCity="testCity" name="test name" image="test.jpg"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});