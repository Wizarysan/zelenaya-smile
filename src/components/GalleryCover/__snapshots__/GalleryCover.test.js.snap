exports[`test renders correctly with test props 1`] = `
<div
  className="gallery_cover text-center">
  <a
    onClick={[Function]}
    style={Object {}}>
    <div
      className="text-left gallery_cover__textwrap">
      <div
        className="js-play-sound"
        onMouseEnter={[Function]}>
        
      </div>
      <div
        className="gallery_cover__hover photoshadow-grey">
        <img
          className="gallery_cover__image"
          src="test.jpg" />
        <div
          className="gallery_cover__hover-green">
          <div
            className="gallery_cover__hover-photos center-both">
            <div
              className="gallery_cover__hover-digit" />
            <div
              className="gallery_cover__hover-text">
              фото
            </div>
          </div>
        </div>
      </div>
      <br />
      <span
        className="gallery_cover__name">
        test name
        <svg
          className="arrow_right"
          viewBox="0 0 24 14.03">
          <path
            d="M579.32,1051.82a0.645,0.645,0,0,0,.894,0l6.6-6.4a0.605,0.605,0,0,0,0-.87l-6.6-6.4a0.645,0.645,0,0,0-.894,0,0.605,0.605,0,0,0,0,.87l5.554,5.38H563.631a0.615,0.615,0,1,0,0,1.23h21.184l-5.5,5.32A0.6,0.6,0,0,0,579.32,1051.82Z"
            transform="translate(-563 -1037.97)" />
        </svg>
      </span>
      <br />
      <span
        className="gallery_cover__date" />
    </div>
  </a>
</div>
`;
